dev/TeachingTip/InteractionTests/TeachingTipTests.cs

```diff
@@ -932,6 +944,72 @@ public void F6PutsFocusOnCloseButton()
             }
         }
 
+        [TestMethod]
+        public void VerifyTeachingTipNotIncludedInTabOrder()
+        {
+            using (var setup = new TestSetupHelper(new[] { "TeachingTip Tests", "TeachingTipFocus Test" }))
+            {
+                var elements = new TeachingTipFocusTestPageElements();
+                var openTeachingTipButton = elements.GetOpenButton();
+
+                Log.Comment("Verify open button has keyboard focus.");
+                Verify.IsTrue(openTeachingTipButton.HasKeyboardFocus);
+
+                Log.Comment("Verify that a closed teaching tip does not appear in tab order.");
+                KeyboardHelper.PressKey(Key.Tab);
+                Wait.ForIdle();
+                Verify.IsTrue(openTeachingTipButton.HasKeyboardFocus);
+
+                Log.Comment("Verify that an opened teaching tip does not appear in tab order.");
+                OpenTeachingTip();
+                Verify.IsTrue(openTeachingTipButton.HasKeyboardFocus);
+                KeyboardHelper.PressKey(Key.Tab);
+                Wait.ForIdle();
+                Verify.IsTrue(openTeachingTipButton.HasKeyboardFocus);
+
+                Log.Comment("Switch to light-dismissable teaching tip.");
+                CloseTeachingTipProgrammatically();
+                EnableLightDismiss();
+
+                Log.Comment("Verify that a closed light-dismissable teaching tip does not appear in tab order.");
+
+                Verify.IsTrue(openTeachingTipButton.HasKeyboardFocus);
+                KeyboardHelper.PressKey(Key.Tab);
+                Wait.ForIdle();
+                Verify.IsTrue(openTeachingTipButton.HasKeyboardFocus);
+
+                void OpenTeachingTip()
+                {
+                    if (elements.GetIsOpenCheckBox().ToggleState != ToggleState.On)
+                    {
+                        openTeachingTipButton.InvokeAndWait();
+                        WaitForChecked(elements.GetIsOpenCheckBox());
+                        WaitForChecked(elements.GetIsIdleCheckBox());
+                    }
+                }
+
+                void CloseTeachingTipProgrammatically()
+                {
+                    if (elements.GetIsOpenCheckBox().ToggleState != ToggleState.Off)
+                    {
+                        elements.GetCloseButton().InvokeAndWait();
+                        WaitForUnchecked(elements.GetIsOpenCheckBox());
+                        WaitForChecked(elements.GetIsIdleCheckBox());
+                    }
+                }
+
+                void EnableLightDismiss()
```

Comment:
>EnableLightDismiss [](start = 21, length = 18)

Aren't there already helpers to do this in this file?

Reply:
Is this because its the only test on the new page?

---
In reply to: [536416017](https://github.com/microsoft/microsoft-ui-xaml/pull/3703#discussion_r536416017) [](ancestors = 536416017)

Reply:
Yes, there are already such methods but they work with the UI elements of TeachingTip's main test page (they use a different "elements" variable to access test UI elements - of type `TeachingTipTestPageElements`.)

Since we only have two teaching tip test pages for now and the second test page added as part of this PR only has one test currently (and I suspect this test will remain an outlier for some time with the majority of the future tests likely to be added to the main test page) I opted for some "quick code" here. As in, just create these helper methods local to this test instead of a potentially broader refactoring to efficiently reuse some code (i.e. passing a TeachingTipTestPageElements instance to the existing EnableLightDismiss() method and perhaps having that instance implement an interface which is shared among test pages to reflect duplicated UI elements).

If in the future we have more tests spread out across multiple test pages and those test pages share a common UI (like an IsOpen status checkbox) then there could be an incentive to look into code sharing between these different test UIs.) For now I think it's fine this way.

Reply:
agreed although it might be worth having the test in a different file.

Reply:
Good idea. We have a separate [FocusBehaviorTests] file for the Navigation. Feels like we could move this test and the existing focus behabior tests (including the interaction tets being added as part of PR https://github.com/microsoft/microsoft-ui-xaml/pull/3545) into such a file for the teaching tip.

Since I have to merge master into PR https://github.com/microsoft/microsoft-ui-xaml/pull/3545/ anyway to resolve the merge conflicts, I could move this test and rthe existing focus tests into a separate FocusBehaviorTests file as part of PR https://github.com/microsoft/microsoft-ui-xaml/pull/3545 - or create a separate PR if that one will take a bit longer to be merged.